pkg/lib/cockpit-components-terminal.jsx

```diff
@@ -316,10 +316,12 @@ export class Terminal extends React.Component {
         const padding = 10; // Leave a bit of space around terminal
         const realHeight = this.terminal._core._renderService.dimensions.css.cell.height;
         const realWidth = this.terminal._core._renderService.dimensions.css.cell.width;
-        if (realHeight && realWidth && realWidth !== 0 && realHeight !== 0)
+        const parentHeight = this.terminalRef.current.parentElement.clientHeight;
+        const parentWidth = this.terminalRef.current.parentElement.clientWidth;
+        if (realHeight && realWidth && realWidth !== 0 && realHeight !== 0 && parentHeight > padding && parentWidth > padding)
```

Comment:
the `- 12` for scrollbar is missing, no?

Reply:
I wasn't sure of this. The *practical* bug is just when parent* are 0. I don't know if you can make a terminal window which is < 22 x 10 pixels (in which case the calculation would still underflow). So either I change the comparisons to `> 0` or include the 12 with `> padding + 12`, but this feels repetitive and conceptually wrong. 

We could just make the whole thing log an error with too small dimensions and fall back to rows=1/colums=1 then.

So * shrug* I don't know, do you have a preference/better idea?

Reply:
Cleanest would probably be to compute the `rows` and `cols` values completely, and then skipping the return when one of them is negative. Or just clipping them to be >= 1.

Reply:
Reworked to do the clipping.